File: app/javascript/channels/modbus_channel.js
import consumer from "channels/consumer"

consumer.subscriptions.create("ModbusChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log("Connected to the modbus channel");
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    console.log(data);
  }
});

File: app/channels/modbus_channel.rb
class ModbusChannel < ApplicationCable::Channel
  def subscribed
    puts "##### connected"
    stream_from 'modbus_channel'
  end

  def unsubscribed
    puts "##### disconnected"

    # Any cleanup needed when channel is unsubscribed
  end

  def modbus(data)
    ActionCable.server.broadcast("modbus_channel",data)
  end
end
